File: c/inout.c
// @expect verified

#include "smack.h"//"assert.h"
#include "stdio.h"

void addFive(int * p) {
    *p = *p + 5;
}

int main(void) {
    int * x = malloc(sizeof(int));
// when the malloc isn't there, this test does not pass, even though clang does the proper behavior when the program runs.
// ^^ 5/14/2018 not sure if this comment is old...
    *x = 2;
    addFive(x);
    int y = *x;
    assert(y == 7);
    return 0;
}

File: objc/inout.m
// @expect verified

#import <Foundation/Foundation.h>
#include "smack.h"

@interface Updater : NSObject

- (void) addFive:(int *)x;

@end

@implementation Updater

- (void) addFive:(int *)x
{
    *x = *x + 5;
}

@end

int main(void)
{
    Updater *updater = [[Updater alloc] init];

    int * x = malloc(sizeof(int));
    *x = 2;
    [updater addFive:x];
    int y = *x;
    assert(y == 7);
    
    [updater release];
    return 0;
}
